webapp/src/containers/LinkModal/index.tsx

```diff
@@ -50,7 +50,8 @@ const LinkModal = () => {
         setProjectDetails({...projectDetails, project: (e.target as HTMLInputElement).value});
     };
 
-    const onConfirm = useCallback(() => {
+    // Handles on confirming link project
+    const onConfirm = () => {
```

Comment:
Why did you remove the `useCallback`?

Reply:
we don't need any caching here even when the callback dependency which would be `errorState` here is changed we need to perform all the logic present inside this func

so, I don't think it worth using that, what you think?

Reply:
I don't understand. If the dependency `errorState` is not changing, then why do we need to perform all the logic inside the function?

Reply:
my bad, I thought the submit button stays disabled when fields are invalidated so it will be enabled only when errors are resolved and that way we would be performing the logic only when the button is clicked this implies error fields are resolved

added callback, but we can enhance our UI to keep submit button disabled if there is any error in the form and no change is made

@sooraj-shukla you can also do this as this is a good practice, it's useless to keep the button enabled when form already have validation errors